lib/realtime/repo.ex

```diff
@@ -52,41 +52,39 @@ defmodule Realtime.Repo do
   end
 
   @doc """
-  Converts a Postgrex.Result into a given struct
+  Deletes all records for a given query and converts them into a given struct
   """
-  @spec result_to_single_struct({:ok, Postgrex.Result.t()} | {:error, any()}, module()) ::
-          {:ok, struct()} | {:ok, nil} | {:error, any()}
-  def result_to_single_struct({:ok, %Postgrex.Result{rows: [row], columns: columns}}, struct) do
+  @spec del(DBConnection.conn(), Ecto.Queryable.t()) ::
+          {:ok, non_neg_integer()} | {:error, any()}
+  def del(conn, query) do
+    with {:ok, %Postgrex.Result{num_rows: rows}} <- run_delete_query(conn, query) do
+      {:ok, rows}
```

Comment:
nitpick: consider using the more descriptive variable name.

```suggestion
    with {:ok, %Postgrex.Result{num_rows: num_rows}} <- run_delete_query(conn, query) do
      {:ok, num_rows}
```